lib/protobuf/extype/enum_transform.ex

```diff
@@ -0,0 +1,102 @@
+defmodule EnumTransform do
+  @moduledoc """
+  Transforms enums if the enum field extension is present. Accepted values are lowercase, deprefix,
+  and atomize. Atomize is an alias for deprefix and lowercase.
+  """
+
+  require Protobuf.Decoder
+  require Logger
+  import Protobuf.Decoder, only: [decode_zigzag: 1]
+
+  @type transform :: String.t()
+
+  @type type :: {:enum, atom}
+
+  @type value :: atom
+
+  def validate_and_get_transformers!({:enum, _type}, "lowercase"), do: [EnumTransform.Lowercase]
```

Comment:
let's clean this up. tread it as an option list instead of hardcoding permutations.